akka-http-core/src/main/java/akka/http/javadsl/model/HttpMessage.java

```diff
@@ -65,6 +65,22 @@
      */
     <T extends HttpHeader> Iterable<T> getHeaders(Class<T> headerClass);
 
+    /**
+     * An iterable containing the attributes for this message.
+     */
+    Iterable<Object> getAttributes();
+
+    /**
+     * Try to find the first attribute of the given class and return
+     * Optional.of(header), otherwise this method returns an empty Optional
+     */
+    <T> Optional<T> getAttribute(Class<T> attributeClass);
```

Comment:
If we leave it allowing multiple maybe it should rather be `firstAttributeOf` or something that conveys what the docs now say (that'd also align it with the collection api somewhat on the Scala side)?

Reply:
Ah, but header can be multiple and getHeader returns first. Got it. NVM then.

Reply:
yeah, it's somewhat confusing but being consistent with `getHeader` might be worth it anyway